annotation-processor/src/main/java/org/hibernate/validator/ap/checks/annotationparameters/AnnotationMessageCheck.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.checks.annotationparameters;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.regex.Pattern;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.Elements;
+
+import org.hibernate.validator.ap.checks.ConstraintCheckIssue;
+import org.hibernate.validator.ap.util.AnnotationApiHelper;
+import org.hibernate.validator.ap.util.CollectionHelper;
+
+/**
+ * Checks that the message provided as annotation parameter is valid and gives a warning otherwise.
+ *
+ * @author Marko Bekhta
+ */
+public class AnnotationMessageCheck extends AnnotationParametersAbstractCheck {
+
+	private static final String WORDS_SEPARATED_WITH_DOTS = "(\\w)+(\\.(\\w)+)*" ;
+
+	// for dots and no {} around, or one of the {} is missing
+	private static final Pattern MESSAGE_PATTERN = Pattern.compile( WORDS_SEPARATED_WITH_DOTS + "|\\{" + WORDS_SEPARATED_WITH_DOTS + "|" + WORDS_SEPARATED_WITH_DOTS + "\\}" );
+
+	private final Elements elementUtils;
+
+	public AnnotationMessageCheck(AnnotationApiHelper annotationApiHelper, Elements elementUtils) {
+		super( annotationApiHelper );
+		this.elementUtils = elementUtils;
+	}
+
+	@Override
+	protected boolean canCheckThisAnnotation(AnnotationMirror annotation) {
+		return true;
+	}
+
+	@Override
+	protected Set<ConstraintCheckIssue> doCheck(Element element, AnnotationMirror annotation) {
+
+		//check if default message on annotation is correct or not:
+		if ( ElementKind.ANNOTATION_TYPE.equals( element.getKind() ) ) {
```

Comment:
Nice. I'd only suggest to put have two separate check classes: one for checking the default message given at a constraint type definition and another one for checking the messages when using them.